pkg/deployment/deployment.go

```diff
@@ -99,127 +99,6 @@ func Deploy(
-	readyMessage = dataplanev1beta1.ValidateNetworkReadyMessage
-	readyErrorMessage = dataplanev1beta1.ValidateNetworkErrorMessage
-	deployFunc = ValidateNetwork
-	deployName = "ValidateNetwork"
-	deployLabel = ValidateNetworkLabel
-	err = ConditionalDeploy(
-		ctx,
-		helper,
-		obj,
-		sshKeySecret,
-		inventoryConfigMap,
-		status,
-		readyCondition,
-		readyMessage,
-		readyWaitingMessage,
-		readyErrorMessage,
-		deployFunc,
-		deployName,
-		deployLabel,
-		aeeSpec,
-		foundService,
-	)
-
-	if err != nil || !status.Conditions.IsTrue(readyCondition) {
-		return &ctrl.Result{}, err
-	}
-	log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
-
-	// InstallOS
-	readyCondition = dataplanev1beta1.InstallOSReadyCondition
-	readyWaitingMessage = dataplanev1beta1.InstallOSReadyWaitingMessage
-	readyMessage = dataplanev1beta1.InstallOSReadyMessage
-	readyErrorMessage = dataplanev1beta1.InstallOSErrorMessage
-	deployFunc = InstallOS
-	deployName = "InstallOS"
-	deployLabel = InstallOSLabel
-	err = ConditionalDeploy(
-		ctx,
-		helper,
-		obj,
-		sshKeySecret,
-		inventoryConfigMap,
-		status,
-		readyCondition,
-		readyMessage,
-		readyWaitingMessage,
-		readyErrorMessage,
-		deployFunc,
-		deployName,
-		deployLabel,
-		aeeSpec,
-		foundService)
-
-	if err != nil || !status.Conditions.IsTrue(readyCondition) {
-		return &ctrl.Result{}, err
-	}
-	log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
-
-	// ConfigureOS
-	readyCondition = dataplanev1beta1.ConfigureOSReadyCondition
-	readyWaitingMessage = dataplanev1beta1.ConfigureOSReadyWaitingMessage
-	readyMessage = dataplanev1beta1.ConfigureOSReadyMessage
-	readyErrorMessage = dataplanev1beta1.ConfigureOSErrorMessage
-	deployFunc = ConfigureOS
-	deployName = "ConfigureOS"
-	deployLabel = ConfigureOSLabel
-	err = ConditionalDeploy(
-		ctx,
-		helper,
-		obj,
-		sshKeySecret,
-		inventoryConfigMap,
-		status,
-		readyCondition,
-		readyMessage,
-		readyWaitingMessage,
-		readyErrorMessage,
-		deployFunc,
-		deployName,
-		deployLabel,
-		aeeSpec,
-		foundService)
-
-	if err != nil || !status.Conditions.IsTrue(readyCondition) {
-		return &ctrl.Result{}, err
-	}
-	log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
-
-	// RunOS
-	readyCondition = dataplanev1beta1.RunOSReadyCondition
-	readyWaitingMessage = dataplanev1beta1.RunOSReadyWaitingMessage
-	readyMessage = dataplanev1beta1.RunOSReadyMessage
-	readyErrorMessage = dataplanev1beta1.RunOSErrorMessage
-	deployFunc = RunOS
-	deployName = "RunOS"
-	deployLabel = RunOSLabel
-	err = ConditionalDeploy(
-		ctx,
-		helper,
-		obj,
-		sshKeySecret,
-		inventoryConfigMap,
-		status,
-		readyCondition,
-		readyMessage,
-		readyWaitingMessage,
-		readyErrorMessage,
-		deployFunc,
-		deployName,
-		deployLabel,
-		aeeSpec,
-		foundService)
-
-	if err != nil || !status.Conditions.IsTrue(readyCondition) {
-		return &ctrl.Result{}, err
-	}
-	log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
-
 	// ConfigureCephClient
```

Comment:
@fultonj does ceph need to be installed specifically before/after some service?

Reply:
@fao89 `ConfigureCephClient` needs to be done before we create the NovaExternal CR.